analyzers/src/SonarAnalyzer.Common/Rules/ClassShouldNotBeEmptyBase.cs

```diff
@@ -51,13 +53,17 @@ public abstract class ClassShouldNotBeEmptyBase<TSyntaxKind, TDeclarationSyntax>
                     && IsEmptyAndNotPartial(c.Node)
                     && !HasAnyAttribute(c.Node)
                     && !HasConditionalCompilationDirectives(c.Node)
+                    && !ShouldIgnoreBecauseOfName(identifier)
                     && !ShouldIgnoreBecauseOfBaseClassOrInterface(c.Node, c.SemanticModel))
                 {
                     c.ReportIssue(Diagnostic.Create(Rule, identifier.GetLocation(), DeclarationTypeKeyword(c.Node)));
                 }
             },
             Language.SyntaxKind.ClassAndRecordClassDeclarations);
 
+    private static bool ShouldIgnoreBecauseOfName(SyntaxToken identifier) =>
+        IgnoredSuffixes.Any(identifier.ValueText.EndsWith);
```

Comment:
Why did you choose `.EndsWith` specifically? I guess to reduce the FN rate

Reply:
What else could I use here?

Reply:
`.Contains`?

Reply:
That would not fit the ticket's specification.